flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java

```diff
@@ -368,29 +363,31 @@ private int updateNpmLocalDependency(JsonObject dependenciesObject,
     }
 
     /**
-     * Compares vaadin-shrinkwrap dependency version (which is the same as
-     * platform version) from the {@code dependencies} object with the current
-     * vaadin-shrinkwrap version (retrieved from file system: package.json,
-     * package-lock.json). In case there was no existing shrinkwrap version,
-     * then version is considered updated.
+     * Compares current platform version with the one last recorded as installed
+     * in node_modules/.vaadin/vaadin_version. In case there was no existing
+     * platform version recorder and node_modules exists, then platform is
+     * considered updated.
      *
-     * @param dependencies
-     *            dependencies object with the vaadin-shrinkwrap version
      * @return {@code true} if the version has changed, {@code false} if not
      * @throws IOException
      *             when file reading fails
      */
-    private boolean isPlatformVersionUpdated(JsonObject dependencies)
-            throws IOException {
-        String shrinkWrapVersion = null;
-        if (dependencies.hasKey(SHRINK_WRAP)) {
-            shrinkWrapVersion = dependencies.getString(SHRINK_WRAP);
+    private boolean isPlatformVersionUpdated() throws IOException {
+        // if no record of current version is present, version is not
+        // considered updated
+        Optional<String> platformVersion = Platform.getVaadinVersion();
+        if (platformVersion.isPresent() && nodeModulesFolder.exists()) {
+            JsonObject vaadinJsonContents = getVaadinJsonContents();
+            // If no record of previous version, version is considered updated;
+            if (vaadinJsonContents == null
+                    || !vaadinJsonContents.hasKey(NodeUpdater.VAADIN_VERSION)) {
```

Comment:
```suggestion
            if (!vaadinJsonContents.hasKey(NodeUpdater.VAADIN_VERSION)) {
```
Null check is not needed as we always get a JsonObject.

Reply:
Thanks, missed it. Done.